docs/topics/lombok.md

```diff
@@ -0,0 +1,137 @@
+If you generate these declarations in a Java module and use them in another Kotlin module, 
+then you don't need to use this plugin.
+
+[Lombok](https://projectlombok.org/) generates Java code itself, not the Kotlin Lombok compiler plugin. 
+Thus, you still need to configure Lombok as usual, when using this plugin. 
+Learn more about [how to make the plugin seeing Lombok's config](#defining-the-place-of-lombok-config).
+
+## Support of annotations
+
+See which annotations the plugin supports and the
+[current state of Lombok compiler plugin](https://github.com/JetBrains/kotlin/blob/master/plugins/lombok/lombok-compiler-plugin/README.md).
+
+Currently, we don't have plans on supporting the @Builder annotation. However, we can consider this if you vote
+for [`@Builder` in YouTrack](https://youtrack.jetbrains.com/issue/KT-46959).
+
+> Kotlin compiler ignores Lombok annotations if you use them in Kotlin code.
+>
+{type="note"}
+
+## Gradle
+
+### Main settings
+
+Apply the `kotlin-plugin-lombok` Gradle plugin in the `build.gradle.kts` file:
+
+```kotlin
+plugins {
+    id ("org.jetbrains.kotlin.plugin.lombok") version "%kotlinVersion%"
+}
+```
+
+Set up the Lombok itself as the plugin:
+
+```kotlin
+plugins {
+    id ("io.freefair.lombok") version "5.3.0"
+}
+```
+
+See the [test project with examples of Lombok compiler plugin usage](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/nokapt).
+
+### Defining the place of lombok.config
+
+If you use the file `lombok.config` for [configuring Lombok](https://projectlombok.org/features/configuration),
+provide a path to it to the plugin. Add the following code to your `build.gradle.kts` file:
+
+```kotlin
+kotlinLombok {
+    lombokConfigurationFile(file("lombok.config"))
+}
+```
+
+See the [test project with examples of Lombok compiler plugin and `lombok.config` usages](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/withconfig).
+
+## Maven
+
+To use Lombok compiler plugin, add the following lines to the `pom.xml` file:
+
+```xml
+<plugin>
+    <groupId>org.jetbrains.kotlin</groupId>
+    <artifactId>kotlin-maven-plugin</artifactId>
+    <version>${kotlin.version}</version>
+    <configuration>
+        <compilerPlugins>
+            <plugin>lombok</plugin>
+        </compilerPlugins>
+        <pluginOptions>
+            <option>lombok:config=${project.basedir}/lombok.config</option>
+        </pluginOptions>
+    </configuration>
+    <dependencies>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-maven-lombok</artifactId>
+            <version>${kotlin.version}</version>
+        </dependency>
+    </dependencies>
+</plugin>
+```
+
+Please refer to the [Gradle](#gradle) section for detailed information about the Lombok compiler plugin's configuration.
```

Comment:
I don't think that the Gradle instruction will actually help... It doesn't show what to write in maven.

Reply:
True. I've added more details, you'll see them in the PR. But maybe we should delete this description at all and leave pure XML code as it is on other plugins pages.